### ISSUE TITLE
MovieClip HTML5 play/stop not working

### ISSUE THREAD

**Issue by KnightMearh** [openfl/openfl#1552]
In the HTML5 target play(), stop(), gotoAndPlay(), gotoAndStop() don't work. I have a MovieClip that has a main timeline with frame labels:

```
var future = AssetLibrary.loadFromFile ("myswf.bundle");
future.onComplete (library_onComplete);

//function library_onComplete
var clip = library.getMovieClip ("");
addChild (clip);
clip.gotoAndStop( "SomeFrameLabel" );
clip.stop();
```

The clip just plays and won't listen to either stop call. Upon investigating the MovieClip code, I see that `__playing` is initially null and only gets set using `play()` and `stop()`. But these functions first check if `__playing` is false or true before toggling. If I trace `__playing` in enter frame, it seems to flip (maybe when the playhead crosses into a new frame label, hard to tell) between true/false/null values. If `__playing` is initialized in the constructor to true, then it seems to get the code stuck in an infinite loop and freezes the browser tab (Chrome). So the issue here is that play() and stop() don't actually cause the clip to play or stop.

target: HTML5
openfl: 4.9.2
lime: 4.0.3

**Comment by jgranick** [comment]
On the latest development version of OpenFL, this appears to be working. I recreated your code like this:

```haxe
package;


import openfl.Assets;
import openfl.display.Sprite;
import openfl.utils.AssetLibrary;


class Main extends Sprite {
	
	
	public function new () {
		
		super ();
		
		var future = AssetLibrary.loadFromFile ("assets/library.bundle");
		future.onComplete (library_onComplete);
		
	}
	
	
	private function library_onComplete (library:AssetLibrary):Void {
		
		var clip = library.getMovieClip ("");
		addChild (clip);
		clip.gotoAndStop( "SomeLabel" );
		clip.stop();
		
	}
	
	
}
```

![capture](https://cloud.githubusercontent.com/assets/833997/25188216/2c7e1378-24da-11e7-8e24-a5d5998b3642.PNG)

Both the `gotoAndStop` and `stop` appear to work fine, I used `openfl process library.swf` to create "library.bundle"

**Comment by jgranick** [comment]
If you use the `myswf.bundle` method, make sure that you run `openfl process myswf.swf` again when changes are made.

You can also use `<library path="Assets/myswf.swf" />` and run `Assets.loadLibrary ("myswf.swf");` instead to have the tools process your SWF automatically when the modification time changes

**Comment by KnightMearh** [comment]
I think I've found the issue. My code / setup was pretty much identical to yours, but I isolated a case that breaks it and it has to do with Animate. If you export your SWF from the stage (main timeline) directly, using `File -> Export Swf`, then you are correct and everything works as intended (play/stop and gotoAndPlay/Stop). However, if you export from the library, `right click on Movie Clip symbol -> Export SWF` then the loaded swf will not behave. Both cases will produce an identical `library.json` file but the `swflite.dat` will have lots of differences depending which export method is used.

Ideally these should produce identical SWFs, so I'm not sure what differs under the hood here.

Thank you for your response earlier, it was helpful to let me narrow a lot of things out.
